validator.go

```diff
@@ -0,0 +1,267 @@
+package jwt
+
+import (
+	"crypto/subtle"
+	"time"
+)
+
+// Validator is the core of the new Validation API. It can either be used to
```

Comment:
Something about using this API just doesn't feel idiomatic and it feels like a bit of work to get a parser with leeway support.

```
parser := jwt.NewParser(
	jwt.WithValidator(
		jwt.NewValidator(
			jwt.WithLeeway(time.Minute),
		),
	),
)
```

> .. or used standalone to validate an already parsed [Claim]

Does the Validator need to be exposed externally? What is the use case for a standalone validator?

Reply:
> Something about using this API just doesn't feel idiomatic and it feels like a bit of work to get a parser with leeway support.
> 
> ```
> parser := jwt.NewParser(
> 	jwt.WithValidator(
> 		jwt.NewValidator(
> 			jwt.WithLeeway(time.Minute),
> 		),
> 	),
> )
> ```

You are right, something like this might be easier I guess:
```golang
parser := jwt.NewParser(jwt.WithLeeway(time.Minute))
```

> 
> > .. or used standalone to validate an already parsed [Claim]
> 
> Does the Validator need to be exposed externally? What is the use case for a standalone validator?

Someone in the previous comments mentioned something about different validation profiles, so not entirely sure. I would like to keep the option though.

I will play around a little bit whether I can include validation options directly in the `NewParser`.

Reply:
What do you think about moving the validator to an `internal` package, and exposing it in the public API if users request it? It's easy to expose, but once it's part of the public API then it's not possible to remove.

This is the only part I'm struggling with.

I'm a bit worried decoupling parsing/verification and validation will cause confusion.

Reply:
With the recent changes, I can just unexport the `validator` struct for now and we can see whether there is additional demand to make it public.

Reply:
Thanks for making that change 🙏